pkg/logs/config/source.go

```diff
@@ -30,6 +33,7 @@ func NewLogSource(name string, config *LogsConfig) *LogSource {
 		inputs:   make(map[string]bool),
 		lock:     &sync.Mutex{},
 		Messages: NewMessages(),
+		Parser:   parser.NewNoopParser(),
```

Comment:
I'm still not 100% sold on this.. I liked the fact that LogSource was a rather descriptive object and not tied to implementation details like how we parse logs. Have you considered other alternatives, like only setting a type and letting the line handler pick a parser (or none) depending on it ?